Removed object adapter hold support

bentoi opened the pull request.

bernardnormier commented on csharp/src/Ice/ConnectionFactory.cs:
```diff
@@ -1581,33 +1504,25 @@ private void SetState(int state)
+                case StateInactive:
                     {
-                        if (_state != StateActive) // Can only switch from active to holding.
+                        if (_state != StateActive) // Can only switch from active to inactive.
```
This is not correct. Presumably the factory's state can only progress from Inactive to Active to Closed to Finished.
And we actually call this private SetState only 3 times, and never with the StateInactive parameter - so this code is never reached.

bentoi replied:
I fixed this.

bernardnormier commented on scripts/tests/Ice/background.py:
```diff
@@ -2,4 +2,5 @@
 #
 
-TestSuite(__name__, libDirs=["testtransport"], options={ "mx": [False] })
+# SSL doesn't work with this test when holding on reading on the server side
```
It's not clear what "holding" means here. Is it possible to run a subset of this test with ssl, and skip a portion of the test based on the transport?

bentoi replied:
The transport wrapper implemented by the test holds on reads: it doesn't call read on the underlying transport. It's possible we could make it work with SSL by disabling some tests but I'd rather not spend time on a test that most likely will have to be removed or completely reworked after the transport API refactoring.

bentoi commented on csharp/test/Ice/acm/AllTests.cs:
```diff
@@ -425,27 +407,25 @@ class CloseOnIdleAndInvocationTest : TestCase
                 base("close on idle and invocation", com, helper)
             {
-                SetClientACM(3, 3, 0); // Only close on idle and invocation
+                SetClientACM(1, 3, 0); // Only close on idle and invocation
```
I'd rather fix this with another PR.

bentoi commented:
I've fixed the OA activate method and further simplified the incoming connection factory. I removed the check for the FD limit error condition (which previously deactivated the acceptor for 1s). 

Waiting for this error condition to occur isn't great as many things will misbehave when it's hit, a better solution would be to allow the user to configure a limit on the number of connections the OA can accept... or the user should do this through firewall for public facing servers.
